pycbc/inference/sampler_kombine.py

```diff
@@ -0,0 +1,196 @@
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides classes and functions for using different sampler
+packages for parameter estimation.
+"""
+
+import numpy
+from pycbc.inference.sampler import _BaseMCMCSampler
+from pycbc.io import WaveformArray, FieldArray
+from pycbc.filter import autocorrelation
+
+#
+# =============================================================================
+#
+#                                   Samplers
+#
+# =============================================================================
+#
+
+class KombineSampler(_BaseMCMCSampler):
+    """This class is used to construct the MCMC sampler from the kombine
+    package.
+
+    Parameters
+    ----------
+    likelihood_evaluator : likelihood class
+        An instance of the likelihood class from the
+        pycbc.inference.likelihood module.
+    nwalkers : int
+        Number of walkers to use in sampler.
+    transd : bool
+        If True, the sampler will operate across parameter spaces using a
+        kombine.clustered_kde.TransdimensionalKDE proposal distribution. In
+        this mode a masked array with samples in each of the possible sets of
+        dimensions must be given for the initial ensemble distribution.
+    processes : {None, int}
+        Number of processes to use with multiprocessing. If None, all available
+        cores are used.
+    min_burn_in : {None, int}
+        Set the minimum number of burn in iterations to use. If None,
+        `burn_in_iterations` will be initialized to `0`.
+    """
+    name = "kombine"
+
+    def __init__(self, likelihood_evaluator, nwalkers, transd=False,
+            processes=None, min_burn_in=None):
+        try:
+            import kombine
+        except ImportError:
+            raise ImportError("kombine is not installed.")
+
+        # construct sampler for use in KombineSampler
+        ndim = len(likelihood_evaluator.waveform_generator.variable_args)
+        sampler = kombine.Sampler(nwalkers, ndim, likelihood_evaluator,
+                                          transd=transd, processes=processes)
```

Comment:
__[flake8]__
* [E127] continuation line over-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->